### PR TITLE
[heft] Throw if jest config file doesn't exist

### PR DESCRIPTION
**Pull request by mmkal**
Fixes https://github.com/microsoft/rushstack/issues/2209

CC @iclanton 

I couldn't find any low-level unit tests for `JestPlugin`, so I created a test file. It doesn't feel quite right but this isn't a blocker for me so I won't be able to put much more time into it I'm afraid. Feel free to push to the branch/close if you think a different approach would be better.

### CODE REVIEW

File: apps/heft/src/plugins/JestPlugin/JestPlugin.ts [hunk 1]
```diff
@@ -22,17 +22,21 @@ export class JestPlugin implements IHeftPlugin {
...
-          await this._runJestAsync(heftSession, heftConfiguration, test);
-        });
-      });
+    const expectedConfigPath: string = path.join(heftConfiguration.buildFolder, JEST_CONFIGURATION_LOCATION);
 
-      heftSession.hooks.clean.tap(PLUGIN_NAME, (clean: ICleanStageContext) => {
-        this._includeJestCacheWhenCleaning(heftConfiguration, clean);
-      });
+    if (!FileSystem.exists(expectedConfigPath)) {
+      throw new Error(`Expected to find jest config file at ${expectedConfigPath}`);
```
Comment (iclanton) [inline apps/heft/src/plugins/JestPlugin/JestPlugin.ts:28]: This should go inside the `_runJestAsync` function, and it should emit an error through the `ScopedLogger`.

**Review by iclanton — CHANGES_REQUESTED** [review]

**Review by iclanton — APPROVED** [review]
Looks good. Thanks!

**Comment by mmkal** [comment]
@iclanton it looks like this found a few projects which are not running tests. Not sure if that's known/expected, or what you think the best course of action is - use a previous heft version for those projects?

**Comment by iclanton** [comment]
I just changed the verb from `test` to `build` for those projects.

**Pull request merged**